manifests/cli.pp

```diff
@@ -23,9 +23,8 @@
   }
 
   $jar = "${jenkins::libdir}/jenkins-cli.jar"
-  $extract_jar = "jar -xf ${jenkins::libdir}/jenkins.war WEB-INF/jenkins-cli.jar"
-  $move_jar = "mv WEB-INF/jenkins-cli.jar ${jar}"
-  $remove_dir = 'rm -rf WEB-INF'
+  $download_jar = 'wget https://localhost/jnlpJars/jenkins-cli.jar -O /tmp/jenkins-cli.jar'
```

Comment:
Downloading to a fixed location in a world writable directory is a security risk. I'd be safer to use `${jar}.downloading` and then move it to `${jar}`. An trivial benefit is that you're also guaranteed that it's on the same filesystem so moving is faster.

Reply:
Also jenkins port is configurable